### DISCUSSION TITLE
Improving the UX for docs

### DISCUSSION THREAD

**Discussion by prjctimg** [prjctimg/huetiful#62] [Announcements]
A lot of effort needs to be put into improving the overall UX of the documentation site. 

Over the next month we'll be adding small but noticeable improvements to help make it easier to navigate the site. Contributions are greatly appreciated devs!!

**Comment by prjctimg** [comment]
Any suggestions @adarshswaminath ?

**Reply by adarshswaminath** [reply]
That sounds like a fantastic initiative! Improving the UX of the documentation site is crucial for providing a better user experience. I'd love to contribute and help make these improvements. Could you please share more details about the specific areas you're planning to work on or any particular tasks that need attention? This way, I can jump in and start working on the areas where my skills can be most beneficial. Let's collaborate to enhance the site's usability and accessibility!

**Reply by prjctimg** [reply]
Great! The optimism and energy will push us beyond 👽!


 I've added some things I think need attention at the moment and still updating the list. The documentation site uses Nextjs and TailwindCSS. You can go to  [timlrx](https://github.com/timlrx/overview) and click on the "nextjs-starter-template" to familiarise with how it works.

Of course, if you have simpler alternative, please share with us so that we work smart not hard!

**Comment by prjctimg** [comment]
@adarshswaminath 

The entire documentation site needs touch ups here and there so I have tried to split up the tasks into parts Homepage, Reading View, and the ListView. 


The palette could be changed to one with cooler colors like blue and the components could use a bit more border radius to give a relaxed feel. 


I'll break down the tasks over the following comments

**Comment by prjctimg** [comment]
###  Homepage

People like what they see so a pretty landing page is a necessary aesthetic. I think they're plenty of landing page examples out there for libraries but here's what I had in mind:
-  A flowfield coverpage that has shifting colors (kinda makes the project goal obvious that its about color). This could be preferably full viewport.
-  A description of the library capabilities, a getting started section and at the footer we can put some statistics about the project e.g stargazers, forks and contributors.


![Screenshot_20231103_124108_com.sec.android.app.sbrowser.jpg](https://github.com/prjctimg/huetiful/assets/104896135/4bf148db-f23e-47c7-8553-4399ce853cbd)



I've attached an example landing page above. Feel free to make suggestions

**Reply by adarshswaminath** [reply]
![image](https://github.com/prjctimg/huetiful/assets/69946307/75ccf69f-9dd0-4ac5-a63b-f85ff40c87ce)

I have created a simple hero section for the documentation on the 'feat/homepage' branch, which can be found at https://github.com/prjctimg/huetiful-docs/pull/2. Any suggestions are welcome

**Reply by prjctimg** [reply]
> ![image](https://github.com/prjctimg/huetiful/assets/69946307/75ccf69f-9dd0-4ac5-a63b-f85ff40c87ce)
> 
> I have created a simple hero section for the documentation on the 'feat/homepage' branch, which can be found at https://github.com/prjctimg/huetiful-docs/pull/2. Any suggestions are welcome

@adarshswaminath 

Looks promising already! Especially the Call to actions will definitely improve the amount of clicks on the site!

I was thinking we could remove that default logo on the top left corner


And have we agreed on a primary color ? Feel free to pick any of your choice!


And then that heading that says "Latest" could be changed to something more contextually relevant like "Utilities/API" . 


And do you have any ideas for the background of the hero section? An animated gradient ? A flowfield ? Checkout [this resource](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://news.ycombinator.com/item%3Fid%3D29378522&ved=2ahUKEwjFrqLtjqyCAxXaiVwKHWOvB2EQFnoECBIQAQ&usg=AOvVaw1G3kAYvvTlGCL4ZjJloh0Q) on flowfields I found! 

Feel free to add any suggestions!

**Reply by adarshswaminath** [reply]
@prjctimg Do you have any suggestions or references for an animated gradient hero section?

**Reply by adarshswaminath** [reply]
![Screenshot from 2023-11-06 06-25-23](https://github.com/prjctimg/huetiful/assets/69946307/786ed30f-311a-4901-a98b-605b00a97e4f)

- [x] Remove that default logo on the top left corner
- [x] Heading that says "Latest" could be changed to something more contextually relevant like "Utilities/API"
- [ ]  ideas for the background of the hero section? An animated gradient ?

Please review the pull request https://github.com/prjctimg/huetiful-docs/pull/3

**Reply by prjctimg** [reply]
> @prjctimg Do you have any suggestions or references for an animated gradient hero section?

@adarshswaminath @foss2cyber 


Apparently it appears that you're working on the same task which is great! 

Maybe you could agree on which approach to use on the hero section; flowfield or animated background gradient.  [Here's a post on that topic for reference](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://bashooka.com/coding/30-css-animated-gradient-examples/&ved=2ahUKEwjH05aDuK6CAxVhm1wKHVxlCboQFnoECBcQAQ&usg=AOvVaw1-Pd2tITNHMWCGqmjgKtcZ)

**Reply by prjctimg** [reply]
> ![Screenshot from 2023-11-06 06-25-23](https://github.com/prjctimg/huetiful/assets/69946307/786ed30f-311a-4901-a98b-605b00a97e4f)
> 
> - [x] Remove that default logo on the top left corner
> - [x] Heading that says "Latest" could be changed to something more contextually relevant like "Utilities/API"
> - [ ]  ideas for the background of the hero section? An animated gradient ?
> 
> Please review the pull request https://github.com/prjctimg/huetiful-docs/pull/3

@adarshswaminath  Reviewed 👍. The UI is taking shape nicely!

**Reply by prjctimg** [reply]
> > @prjctimg Do you have any suggestions or references for an animated gradient hero section?
> 
> @adarshswaminath @foss2cyber 
> 
> 
> Apparently it appears that you're working on the same task which is great! 
> 
> Maybe you could agree on which approach to use on the hero section; flowfield or animated background gradient.  [Here's a post on that topic for reference](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://bashooka.com/coding/30-css-animated-gradient-examples/&ved=2ahUKEwjH05aDuK6CAxVhm1wKHVxlCboQFnoECBcQAQ&usg=AOvVaw1-Pd2tITNHMWCGqmjgKtcZ) 
> 
> 

@takanome-dev Please help with issue assignment and tracking so that we don't bump heads 😂. I'm still new to issue tracking management. There's an issue for this enhancement, right?

**Reply by adarshswaminath** [reply]
@prjctimg @H-K-R @foss2cyber @takanome-dev I've created another pull request in the docs. If it's messy, I apologize—I'm new to being a collaborator

**Reply by takanome-dev** [reply]
> There's an issue for this enhancement, right?

Oh yeah but no worries :sweat_smile: @adarshswaminath can you let @foss2cyber work on the hero section? He seems to be interested. I assigned him the issue https://github.com/prjctimg/huetiful/issues/70

And to avoid conflicts and ensure a smooth collaboration, we may need to have clear processes and guidelines in place for our dev team and contributors like a `CONTRIBUTING` guide. 
But for now, before start working on a feature or bug fixes, let's first open an issue and assign ourself (or a contributor). Then, mention in our pull request what issue it fixes.

@prjctimg feel free to create templates for issues and pull request. 
Learn more about templates:
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates
- https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

Examples:
- https://github.com/takanome-dev/blog-with-astro-golang/blob/main/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/takanome-dev/blog-with-astro-golang/blob/main/.github/ISSUE_TEMPLATE/bug.yml

**Reply by prjctimg** [reply]
@takanome-dev Woah 🤯 

This sounds like an optimum workflow. Is there a way we can automate some of these tasks as well ? Such as GitHub Actions? 

Will definitely check out the articles enlisted soon! 


Currently working on a major pull request for issue [#68](https://github.com/prjctimg/huetiful/issues/68)


A lot of testing is coming 👀

**Reply by takanome-dev** [reply]
>  Is there a way we can automate some of these tasks as well ? Such as GitHub Actions?

Yeah we can automate assigning issues with github actions, like we do on [open-sauced](https://github.com/open-sauced). i.e -> https://github.com/open-sauced/app/issues/1988

![2023-11-08_10-04](https://github.com/prjctimg/huetiful/assets/79809121/6cd032c8-8713-4588-9b1f-bac62c23a941)

Here is the workflow file -> https://github.com/open-sauced/hot/blob/beta/.github/workflows/triage.yml

**Reply by prjctimg** [reply]
> >  Is there a way we can automate some of these tasks as well ? Such as GitHub Actions?
> 
> Yeah we can automate assigning issues with github actions, like we do on [open-sauced](https://github.com/open-sauced). i.e -> https://github.com/open-sauced/app/issues/1988
> 
> ![2023-11-08_10-04](https://github.com/prjctimg/huetiful/assets/79809121/6cd032c8-8713-4588-9b1f-bac62c23a941)
> 
> Here is the workflow file -> https://github.com/open-sauced/hot/blob/beta/.github/workflows/triage.yml



Okay, thanks! Will customize it a bit to suit our variables. Will also check their CONTRIBUTING file and see how they did it.

Currently working on ESLint, Prettier and husky config.

**Comment by prjctimg** [comment]
### ReadingView

For the reading view, there a few adjustments that could be made to make the documentation easier to read and navigate. 
-  A reading progress bar that updates as a user scrolls down the document [here is an example blog post on such a feature](https://blog.logrocket.com/creating-reading-progress-bar-react/)
- A Table of Contents Table that lists all the headings in the document. Since the source files for the docs are plain text files `.mdx`  we could use  a remark-plugin or a RegExp that maps through all headings and adds them to the TOC
-  The code blocks scroll horizontally on small screens. It would be nice to add some wrapping add use a lighter monospace font for the code blocks (I think the font for code is a bit too heavy)
- An edit in GitHub button on every page so that its easy to fix typos and adjust the docs. A 'was this doc' useful statistic would also help us better our documentation, but of course its a 'nice to have' feature. 
![Here's an example from the MUI docs:](https://github.com/prjctimg/huetiful/assets/104896135/e2736613-ae9b-42e3-a23f-fb7eddc74858)



- Add a link to the source code for each utility. Since every H4 element in the doc source files is a  heading for a utility we can attach these links by dynamically attaching the heading title to a path pattern to the repository ![Here's an example from the Lodash docs](https://github.com/prjctimg/huetiful/assets/104896135/29230ea3-2ed1-40e4-9a1a-65b5b35ba1a0)


 As you can see above, `source` and `npm package` are links that lead to external resources. In this case we only need to link to the source. Also notice how the details for each component are laid out in a well spaced manner making readability a breeze. All the utilities are documented using a uniform schema  containing the fields:
- Parameters
- Description
- Returns
- Example

Maybe a React component to template this layout would be effecient.

**Comment by prjctimg** [comment]
### ListView

For the list view, there's not much to change except make a few design changes to the basic unit (each post in this case). The list would look more engaging if it had shadows and had each post in card format. Here's an example of what the card could look like. I have taken it from [George Francis's personal blog](https://georgefrancis.dev)
![The list view example](https://github.com/prjctimg/huetiful/assets/104896135/d3b3c77b-f84d-4b9e-b42b-9a9f59aeb35b)
